qa/tasks/cephfs/test_sessionmap.py

```diff
@@ -230,3 +230,44 @@ def test_session_evict_blocklisted(self):
 
         self.mount_a.kill_cleanup()
         self.mount_a.mount_wait()
+
+    def test_session_evict_non_blocklisted(self):
+        """
+        Check that mds evicts without blocklisting client
+        """
+
+        self.fs.set_ceph_conf('mds', 'mds session blocklist on evict', False)
+        self.fs.mds_restart()
+
+        self.fs.set_max_mds(2)
+        status = self.fs.wait_for_daemons()
+
+        self.mount_a.run_shell_payload("mkdir {d0,d1} && touch {d0,d1}/file")
+        self.mount_a.setfattr("d0", "ceph.dir.pin", "0")
+        self.mount_a.setfattr("d1", "ceph.dir.pin", "1")
+        self._wait_subtrees([('/d0', 0), ('/d1', 1)], status=status)
+
+        self.mount_a.run_shell(["touch", "d0/f0"])
+        self.mount_a.run_shell(["touch", "d1/f0"])
+        self.mount_b.run_shell(["touch", "d0/f1"])
+        self.mount_b.run_shell(["touch", "d1/f1"])
+
+        self.assert_session_count(2, mds_id=self.fs.get_rank(rank=0, status=status)['name'])
+        self.assert_session_count(2, mds_id=self.fs.get_rank(rank=1, status=status)['name'])
+
+        mount_a_client_id = self.mount_a.get_global_id()
+        self.fs.mds_asok(['session', 'evict', "%s" % mount_a_client_id],
+                         mds_id=self.fs.get_rank(rank=0, status=status)['name'])
+
+        self.mount_a.run_shell(["touch", "d0/f00"])
+        self.mount_a.run_shell(["touch", "d1/f00"])
+        self.mount_b.run_shell(["touch", "d0/f10"])
+        self.mount_b.run_shell(["touch", "d1/f10"])
```

Comment:
shouldn't this IO be done after line 270? to ensure that the IO happens fine after the client has been reconnected after getting evicted?

Reply:
This will make the clients to send `client requests` to MDS and during that the clients will open new sessions after the previous sessions being evicted.

And then after this the sessions will be `2` in Line#269 and Line#270.

Reply:
got it and without this IO there will be no sessions opened